src/st/no0/5BDCC.c

```diff
@@ -1,8 +1,134 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 #include "common.h"
-#include "game.h"
+#include "no0.h"
 
-INCLUDE_ASM("st/no0/nonmatchings/5BDCC", func_us_801DBDCC);
+#define VASE self + 1
+#define BEAKER self + 2
+#define WALKING_TABLE self + 3
+
+extern u16 D_us_80180C14[];
+extern s16 D_us_801825CC[];
+extern u16 D_us_801825D4[];
+extern u8 D_us_801825F4[];
+
+typedef enum {
+    /* 0 */ OUIJA_TABLE_INIT,
+    /* 1 */ OUIJA_TABLE_INIT_SUBENTITIES,
+    /* 2 */ OUIJA_TABLE_COME_ALIVE,
+    /* 3 */ OUIJA_TABLE_ATTACK,
+    /* 4 */ OUIJA_TABLE_UNUSED,
+    /* 255 */ OUIJA_TABLE_DEBUG = 255,
+} OuijaTableSteps;
+
+// Main Ouija Table entity
+void EntityOuijaTable(Entity* self) {
+    Entity* otherEntity;
+
+    if (self->flags & FLAG_DEAD) {
+        otherEntity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+        if (otherEntity != NULL) {
+            CreateEntityFromEntity(E_EXPLOSION, self, otherEntity);
+            otherEntity->params = 1;
+        }
+        PlaySfxPositional(SFX_STUTTER_FIREBALL);
+        // Destroy the spinning objects on the table
+        otherEntity = VASE;
+        otherEntity->flags |= FLAG_UNK_2000;
+
+        otherEntity = BEAKER;
+        otherEntity->flags |= FLAG_UNK_2000;
+        DestroyEntity(self);
+        return;
+    }
+
+    switch (self->step) {
+    case OUIJA_TABLE_INIT:
+        InitializeEntity(D_us_80180C14);
+        self->animCurFrame = 5;
+        self->hitboxOffY = 2;
+        return;
+    case OUIJA_TABLE_INIT_SUBENTITIES:
+        if (UnkCollisionFunc3(D_us_801825D4) & 1) {
+            // Spawn objects on the table
+            otherEntity = VASE;
+            CreateEntityFromEntity(E_OUIJA_TABLE_COMPONENT, self, otherEntity);
+            otherEntity->params = 0;
+            otherEntity->ext.ouijaTableContents.parent = self;
+            otherEntity->posX.i.hi -= 4;
+            otherEntity->posY.i.hi -= 16;
+
+            otherEntity = BEAKER;
+            CreateEntityFromEntity(E_OUIJA_TABLE_COMPONENT, self, otherEntity);
+            otherEntity->params = 1;
+            otherEntity->ext.ouijaTableContents.parent = self;
+            otherEntity->posX.i.hi += 2;
+            otherEntity->posY.i.hi -= 20;
+
+            otherEntity = WALKING_TABLE;
+            CreateEntityFromEntity(E_OUIJA_TABLE_COMPONENT, self, otherEntity);
+            otherEntity->params = 2;
+            otherEntity->ext.ouijaTableContents.parent = self;
+
+            if (self->params) {
+                otherEntity->posX.i.hi += 24;
+                otherEntity->facingLeft = 1;
+            } else {
+                otherEntity->posX.i.hi -= 24;
+            }
+#ifndef VERSION_PSP
```

Comment:
This gets optimized out so you don't need the ifdef

Reply:
On the PSP scratch it seems to add extra instructions for me?
```
lh      v1,6(s0)
sh      v1,6(s0)
```

Reply:
I mean you can remove the whole block:
```
#ifndef VERSION_PSP
            // Looks like a coding error
            otherEntity->posY.i.hi = otherEntity->posY.i.hi;
#endif
```

Reply:
Nevermind, I tried it again and it introduced reg swaps, made a mistake

Reply:
Removing the block then causes the PSX to mismatch due to missing instructions. 

```
sh      v0,6(s1)
lhu     v0,0x2c(s0)
nop
```

I thought it would be a no-op too but I guess not?